### PR TITLE
Add support for DsrManager 

### PR DESCRIPTION
**Pull request by KentonPrescott**
In this PR, I've wrapped the DsrManager contract, located [here](https://github.com/makerdao/dsr-manager/blob/master/src/DsrManager.sol), added unit test support (all passing) and updated the CdpManager unit tests to improve consistency.

Most public interfaces within the contract are exposed in `DsrManager`.

[DsrManager.daiBalance()](https://github.com/makerdao/dsr-manager/blob/master/src/DsrManager.sol#L89) is an on-chain utility function that calculates an Address's Dai Balance with a fresh `Pot.chi`. Since the value is only `returned` on chain, it's not useful for off-chain applications; thus, `daiBalance()` hasn't been added to pymaker at this time. 

To view the dai balance with pymaker, I've added the `dai_of` function to the `DsrManager` class that uses the latest `Pot.chi`, but because `chi` is not calculated on-chain, the returned Dai balance could be slightly "stale".

### CODE REVIEW

File: pymaker/dsrmanager.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    def dai_adapter(self) -> DaiJoin:
+        address = Address(self._contract.functions.daiJoin().call())
+        return DaiJoin(self.web3, address)
+
+    def supply(self) -> Wad:
+        """Total supply of pie locked in Pot through DsrManager"""
+        return Wad(self._contract.functions.supply().call())
+
+    def pieOf(self, usr: Address) -> Wad:
```
Comment (EdNoepel) [inline pymaker/dsrmanager.py:62]: For consistency with (most of) this API and PEP8, please use [lowercase with underscores](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables) for method names.

File: pymaker/dss.py [hunk 2]
```diff
@@ -929,7 +929,7 @@ def rho(self) -> datetime:
     def drip(self) -> Transact:
         return Transact(self, self.web3, self.abi, self.address, self._contract, 'drip', [])
 
-    """Note that join requires use of proxy contract, so join/exit are not provided here."""
+    """Note that join requires use of proxy contract, so join/exit are located in pymaker/Dsr.py"""
```
Comment (EdNoepel) [inline pymaker/dss.py:932]: This should be `pymaker/dsrmanager.py`, right?

File: tests/test_cdpmanager.py [hunk 3]
```diff
@@ -16,29 +16,26 @@
...
-from tests.conftest import mcd, web3
+from pymaker.deployment import DssDeployment
+from pymaker.cdpmanager import Urn
 
 
 class TestCdpManager:
-    def setup_class(self):
-        dss_deployment = mcd(web3())
-        self.ilk = dss_deployment.collaterals['ETH-A'].ilk
-        self.cdpmanager = CdpManager(web3(), Address("0x84617303947304444Ceb641582c024f277BBF4Ff"))
```
Comment (EdNoepel) [inline tests/test_cdpmanager.py:27]: Much thanks for cleaning this up.

File: tests/test_dsrmanager.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2020 kentonprescott
```
Comment (Lollike) [inline tests/test_dsrmanager.py:3]: The copyright should be changed to Copyright (C) 2020 Maker Ecosystem Growth Holdings, INC.

File: pymaker/dsrmanager.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
+
+# This file is part of Maker Keeper Framework.
+#
+# Copyright (C) 2020 kentonprescott
```
Comment (Lollike) [inline pymaker/dsrmanager.py:4]: The copyright should be changed to Copyright (C) 2020 Maker Ecosystem Growth Holdings, INC.

File: pymaker/dss.py (hunk 2 above)
Comment (KentonPrescott) [reply pymaker/dss.py:932]: It should be both. Changed to `Join/Exit in Pot can be invoked through pymaker/dsrmanager.py and pymaker/dsr.py`

**Pull request merged**